moonstone/filters/__init__.py

```diff
@@ -0,0 +1,5 @@
+from .filtering import (  # noqa
+    Filtering,
+    FilteringBase,
```

Comment:
This is not something we want to provide to the user, the purpose of this class is to be a scaffold to write other Filtering classes. It might be worth switching All `Bases` classes to abstract classes.